taskchampion/taskchampion/src/server/cloud/server.rs

```diff
@@ -0,0 +1,1130 @@
+    }
+
+    #[test]
+    fn add_version_unknown() {
+        let mut server = make_server();
+        let (v1, v2) = (Uuid::new_v4(), Uuid::new_v4());
+        server.mock_add_version(v1, v2, 1000, b"first");
+        server.mock_set_latest(v2);
+        let (res, _) = server
+            .add_version(Uuid::new_v4(), b"history".to_vec())
+            .unwrap();
+        assert_eq!(res, AddVersionResult::ExpectedParentVersion(v2));
+        let mut expected = server.empty_copy();
+        expected.mock_add_version(v1, v2, 1000, b"first");
+        expected.mock_set_latest(v2);
+        assert_eq!(server.unencrypted(), expected.unencrypted());
+    }
+
+    #[test]
+    fn get_child_version_empty() {
+        let mut server = make_server();
+        assert_eq!(
+            server.get_child_version(Uuid::new_v4()).unwrap(),
+            GetVersionResult::NoSuchVersion
+        );
+    }
+
+    #[test]
+    fn get_child_version_single() {
+        let mut server = make_server();
+        let (v1, v2) = (Uuid::new_v4(), Uuid::new_v4());
+        server.mock_add_version(v2, v1, 1000, b"first");
+        assert_eq!(
+            server.get_child_version(v1).unwrap(),
+            GetVersionResult::NoSuchVersion
+        );
+        assert_eq!(
+            server.get_child_version(v2).unwrap(),
+            GetVersionResult::Version {
+                version_id: v1,
+                parent_version_id: v2,
+                history_segment: b"first".to_vec(),
+            }
+        );
+    }
+
+    #[test]
+    fn get_child_version_multiple() {
+        let mut server = make_server();
+        let (v1, v2, v3) = (Uuid::new_v4(), Uuid::new_v4(), Uuid::new_v4());
+        let (vx, vy, vz) = (Uuid::new_v4(), Uuid::new_v4(), Uuid::new_v4());
+        server.mock_add_version(v1, v2, 1000, b"second");
+        server.mock_add_version(v1, vx, 1000, b"false start x");
+        server.mock_add_version(v1, vy, 1000, b"false start y");
+        server.mock_add_version(v2, v3, 1000, b"third");
+        server.mock_add_version(v2, vz, 1000, b"false start z");
+        server.mock_set_latest(v3);
+        assert_eq!(
+            server.get_child_version(v1).unwrap(),
+            GetVersionResult::Version {
+                version_id: v2,
+                parent_version_id: v1,
+                history_segment: b"second".to_vec(),
+            }
+        );
+        assert_eq!(
+            server.get_child_version(v2).unwrap(),
+            GetVersionResult::Version {
+                version_id: v3,
+                parent_version_id: v2,
+                history_segment: b"third".to_vec(),
+            }
+        );
+        assert_eq!(
+            server.get_child_version(v3).unwrap(),
+            GetVersionResult::NoSuchVersion
+        );
+    }
+
+    #[test]
+    fn cleanup_empty() {
+        let mut server = make_server();
+        server.cleanup().unwrap();
+    }
+
+    #[test]
+    fn cleanup_linear() {
```

Comment:
Are we testing that `cleanup()` doesn't delete anything when it shouldn't? A comment might help.